web_client/src/components/WindowWidget.vue

```diff
@@ -53,8 +54,9 @@ export default defineComponent({
     function updateFromRange([v0, v1]) {
       if (windowLocked.value) return;
       if (v0 === currentRange.value[0] && v1 === currentRange.value[1]) return;
+      currentRange.value = [v0, v1];
       const ww = v1 - v0;
-      const wl = v0 + Math.ceil(ww / 2);
+      const wl = v0 + Math.floor(ww / 2);
```

Comment:
Should we just do `Math.round`?

Reply:
The problem I discovered with rounding up here is that it sometimes creates the cyclical update case where the range values are updated by +1 each tick. This happened when the upper limit had any kind of decimal (which happens when the values are computed by changing the window via a click-and-drag in the render). The Math operation would continuously round up the computed value but leave the decimal on the range max. Using floor here ensures that we drop the decimal every time. This was my best attempt, though; I'm not sure it completely solves that problem but I haven't seen it appear again